flink-docs/src/test/java/org/apache/flink/docs/configuration/ConfigOptionsDocsCompletenessITCase.java

```diff
@@ -61,50 +64,105 @@
+				}
+			});
+	}
+
+	private static <O> void assertOptionsAreWellDefined(Map<String, List<O>> allOptions, BiFunction<O, O, AssertionError> duplicateHandler) {
+		allOptions.entrySet().stream()
+			.filter(entry -> !WELL_DEFINED_WHITELIST.contains(entry.getKey()))
+			.map(Map.Entry::getValue)
+			.forEach(options -> options.stream().reduce((option1, option2) -> {
+				if (option1.equals(option2)) {
+					// we allow multiple instances of ConfigOptions with the same key if they are identical
+					return option1;
+				} else {
+					throw duplicateHandler.apply(option1, option2);
+				}
+			}));
+	}
+
+	private static void compareDocumentedAndExistingOptions(Map<String, List<DocumentedOption>> documentedOptions, Map<String, List<ExistingOption>> existingOptions) {
 		final Collection<String> problems = new ArrayList<>(0);
 
 		// first check that all existing options are properly documented
-		existingOptions.forEach((key, supposedState) -> {
-			DocumentedOption documentedState = documentedOptions.remove(key);
-
-			// if nothing matches the docs for this option are up-to-date
-			if (documentedState == null) {
-				// option is not documented at all
-				problems.add("Option " + supposedState.key + " in " + supposedState.containingClass + " is not documented.");
-			} else if (!supposedState.defaultValue.equals(documentedState.defaultValue)) {
-				// default is outdated
-				problems.add("Documented default of " + supposedState.key + " in " + supposedState.containingClass +
-					" is outdated. Expected: " + supposedState.defaultValue + " Actual: " + documentedState.defaultValue);
-			} else if (!supposedState.description.equals(documentedState.description)) {
-				// description is outdated
-				problems.add("Documented description of " + supposedState.key + " in " + supposedState.containingClass +
-					" is outdated.");
+		existingOptions.forEach((key, supposedStates) -> {
+			List<DocumentedOption> documentedState = documentedOptions.remove(key);
+
+			for (ExistingOption supposedState : supposedStates) {
+				if (documentedState == null || documentedState.isEmpty()) {
+					// option is not documented at all
+					problems.add("Option " + supposedState.key + " in " + supposedState.containingClass + " is not documented.");
+				} else if (documentedState.stream().noneMatch(documentedOption -> supposedState.defaultValue.equals(documentedOption.defaultValue))) {
+					// default is outdated
+					problems.add("Documented default of " + supposedState.key + " in " + supposedState.containingClass +
+						" is outdated. Expected: " + supposedState.defaultValue);
+				} else if (documentedState.stream().noneMatch(documentedOption -> supposedState.description.equals(documentedOption.description))) {
+					// description is outdated
+					problems.add("Documented description of " + supposedState.key + " in " + supposedState.containingClass +
+						" is outdated.");
+				} else {
+					// the docs for this option are up-to-date
+				}
```

Comment:
I think we won't fail if we have a `DocumentedOption` which has no corresponding `ExistingOption` since we are only looking whether there is a `DocumentedOption` for every `ExistingOption`.

Reply:
That is checked just below this block by verifying that the set of documented options (effectively the set of _unmatched_ options) is empty.

Reply:
However, something that could indeed slip through if there are say, 3 documented options with key X, but only 2 existing options with that key.

Reply:
Yes, this is what I meant.